NitroxClient/GameLogic/Crafting.cs

```diff
@@ -9,27 +12,453 @@ namespace NitroxClient.GameLogic
         {
-            string crafterGuid = GuidHelper.GetGuid(gameObject);
+#if TRACE && GAMEEVENTBUILDING
+            NitroxModel.Logger.Log.Debug("BuilderTool_Post_OnHover");
+#endif
+            if (constructable.techType == TechType.Fabricator || constructable.techType == TechType.Workbench)
+            {
+                //if Crafter is in use by remote Player > Display
+                _lastHoveredConstructable = constructable;
+                string _crafterGuid = GuidHelper.GetGuid(constructable.gameObject);
+                bool _inUseByOther = false;
+                string _playerName = string.Empty;
+
+                lock (fabricatorsInUseByRemotePlayers)
+                {
+                    _inUseByOther = fabricatorsInUseByRemotePlayers.ContainsKey(_crafterGuid);
+                    if (_inUseByOther)
+                    {
+                        _playerName = fabricatorsInUseByRemotePlayers[_crafterGuid].PlayerName;
+                    }
+                }
+
+                if (_inUseByOther && _crafterGuid != string.Empty)
+                {
+                    string _baseString = "In use by";
+                    //TODO: Translate _baseString
 
-            FabricatorItemPickup fabricatorItemPickup = new FabricatorItemPickup(crafterGuid, techType.Model());
-            packetSender.Send(fabricatorItemPickup);
-            Log.Debug(fabricatorItemPickup);
+                    //
+                    string _displayText = _baseString + " " + _playerName;
+                    HandReticle.main.SetInteractText(_displayText, string.Empty);
+                    HandReticle.main.SetInteractInfo(_displayText, string.Empty); //must be used, otherwise it is not displayed
+                    HandReticle.main.SetIcon(HandReticle.IconType.HandDeny, 1f);
+                }
+            }
+            else
+            {
+                _lastHoveredConstructable = null;
+            }
         }
+
+        public bool BuilderTool_Pre_HandleInput(GameObject gameObject)
+        {
+#if TRACE && GAMEEVENTBUILDING
+            NitroxModel.Logger.Log.Debug("BuilderTool_Pre_HandleInput");
+#endif
+            if (_lastHoveredConstructable != null)
+            {
+                //if Crafter is in use by remote Player, supress deconstruction
+                string _crafterGuid = GuidHelper.GetGuid(_lastHoveredConstructable.gameObject);
+                bool _inUseByOther = false;
+                string _playerName = string.Empty;
+
+                lock (fabricatorsInUseByRemotePlayers)
+                {
+                    _inUseByOther = fabricatorsInUseByRemotePlayers.ContainsKey(_crafterGuid);
+                    if (_inUseByOther)
+                    {
+                        _playerName = fabricatorsInUseByRemotePlayers[_crafterGuid].PlayerName;
+                    }
+                }
+
+                if (_inUseByOther && _crafterGuid != string.Empty)
```

Comment:
The guid shouldn't be empty as GuidHelper should assign it a guid.

Reply:
Yes. This is a relict of having the GetGuid method overloaded, so that it could return an empty string, to see if guid transfering and assigning has properly happend or something went wrong and the guid wasn't already set.